storm-core/src/jvm/org/apache/storm/pacemaker/PacemakerClientPool.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.pacemaker;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import org.apache.storm.Config;
+import org.apache.storm.generated.HBMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PacemakerClientPool {
+
+    private static final Logger LOG = LoggerFactory.getLogger(PacemakerClientPool.class);
+    
+    private ConcurrentHashMap<String, PacemakerClient> clientForServer = new ConcurrentHashMap<>();
+    private ConcurrentLinkedQueue<String> servers;
+    private Map config;
+
+    public PacemakerClientPool(Map config) {
+        this.config = config;
+        List<String> serverList = (List<String>)config.get(Config.PACEMAKER_SERVERS);
+        if(serverList == null) {
+            serverList = new ArrayList<String>();
+        }
+        else {
+            serverList = new ArrayList<String>(serverList);
+        }
+        Collections.shuffle(serverList);
+        if(serverList != null) {
+            servers = new ConcurrentLinkedQueue<String>(serverList);
+        }
+        else {
+            servers = new ConcurrentLinkedQueue<String>();
+        }
+    }
+    
+    public HBMessage send(HBMessage m) throws PacemakerConnectionException {
+        try {
+            return getWriteClient().send(m);
+        } catch (Exception e) {
+            rotateClients();
+            throw e;
+        }
+    }
+
+    public List<HBMessage> sendAll(HBMessage m) throws PacemakerConnectionException {
+        List<HBMessage> responses = new ArrayList<HBMessage>();
+        LOG.info("Using servers: {}", servers);
```

Comment:
Perhaps this should be debug (especially if we are worried about filling the logs)